validator/db/kv/attestation_history_v2.go

```diff
@@ -340,3 +343,89 @@ func (store *Store) shouldMigrateAttestations() (bool, error) {
 	})
 	return importAttestations, err
 }
+
+// HighestSignedSourceEpoch returns the highest signed source epoch for a validator public key.
+// If no data exists, returning 0 is a sensible default.
+func (store *Store) HighestSignedSourceEpoch(ctx context.Context, publicKey [48]byte) (uint64, error) {
+	ctx, span := trace.StartSpan(ctx, "Validator.HighestSignedSourceEpoch")
+	defer span.End()
+
+	var err error
+	var highestSignedSourceEpoch uint64
+	err = store.view(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(highestSignedSourceBucket)
+		highestSignedSourceBytes := bucket.Get(publicKey[:])
+		if len(highestSignedSourceBytes) == 0 {
```

Comment:
Can you return here if len is less than 8? Or maybe return an error.

bytesutil.BytesToUint64BigEndian will return 0 if input is less than 8 bytes.